stubs/urllib3/urllib3/contrib/socks.pyi

```diff
@@ -0,0 +1,41 @@
+from collections.abc import Mapping
+from typing import Any, ClassVar
+from typing_extensions import TypedDict
+
+from ..connection import HTTPConnection, HTTPSConnection
+from ..connectionpool import HTTPConnectionPool, HTTPSConnectionPool
+from ..poolmanager import PoolManager
+
+class _TYPE_SOCKS_OPTIONS(TypedDict):
+    socks_version: int
+    proxy_host: str | None
+    proxy_port: str | None
+    username: str | None
+    password: str | None
+    rdns: bool
+
+class SOCKSConnection(HTTPConnection):
+    def __init__(self, _socks_options: _TYPE_SOCKS_OPTIONS, *args, **kwargs) -> None: ...
+    def _new_conn(self) -> Any: ...  # "socks.socksocket":
+
+class SOCKSHTTPSConnection(SOCKSConnection, HTTPSConnection): ...
+class SOCKSHTTPConnectionPool(HTTPConnectionPool): ...
+class SOCKSHTTPSConnectionPool(HTTPSConnectionPool): ...
+
+class _ConnectionPoolClasses(TypedDict):
+    http: type[SOCKSHTTPSConnectionPool]
+    https: type[SOCKSHTTPSConnectionPool]
+
+class SOCKSProxyManager(PoolManager):
+    pool_classes_by_scheme: ClassVar[_ConnectionPoolClasses]
```

Comment:
This is overridden on instances, so it's wrong to annotate it as a `ClassVar`, despite the fact it has a class-level default: https://github.com/urllib3/urllib3/blob/aa3def7d242525e6e854991247c4b68583d15135/src/urllib3/contrib/socks.py#L216

```suggestion
   # has a class-level default, but is overridden on instances, so not a ClassVar
    pool_classes_by_scheme: _ConnectionPoolClasses
```